### PR TITLE
erlang: Update to 24.2

### PR DESCRIPTION
**Pull request by secsaba**
#### Description

Update to 24.2

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 12.1 21C52 x86_64
Xcode 13.2.1 13C100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/erlang/Portfile [hunk 1]
```diff
@@ -5,8 +5,8 @@ PortGroup           wxWidgets 1.0
 PortGroup           compiler_blacklist_versions 1.0
 
 name                erlang
-version             23.1
-revision            3
+version             24.2
+revision            1
```
Comment (herbygillot) [inline lang/erlang/Portfile:9]: Revision resets to 0 on version update?

File: lang/erlang/Portfile [hunk 2]
```diff
@@ -42,30 +42,25 @@ distfiles           otp_src_${version}${extract.suffix}                    \
...
-                    size    89063191 \
-                    otp_doc_man_23.1.tar.gz \
-                    rmd160  6d34113ccda6de7b1ca68149bba40420f8726e3b \
-                    sha256  f49ecbb05b0895ec54fc66ea65b4e7cf593a4877748acc4d549f7613c92b94a9 \
-                    size    1384042 \
-                    otp_doc_html_23.1.tar.gz \
-                    rmd160  2a51abbe2a1da25750738628da66a11e16e95afa \
-                    sha256  0e0075f174db2f9b5a0f861263062942e5a721c40ec747356e482e3be2fb8931 \
-                    size    36285075
+checksums           otp_src_24.2.tar.gz \
```
Comment (herbygillot) [inline lang/erlang/Portfile:45]: This would be better written using `${version}` so that this doesn't need to be changed on every update:

```
checksums           otp_src_${version}.tar.gz \
...
                    otp_doc_man_${version}.tar.gz \
...
                    otp_doc_html_${version}.tar.gz \
...
```

**Comment by herbygillot** [comment]
Thanks for this update @secsaba, just some minor comments.

**Comment by evanmiller** [comment]
Might be worth addressing https://trac.macports.org/ticket/62517 while you are at it

**Comment by cardiex-jgreene** [comment]
I just switched from Homebrew to Macports (because Homebrew taking over /opt/local is messing up my multi-user system) and first thing I did was try to install RabbitMQ and ran face-first into Erlang breaking on my M1 chip Mac. I dug around Trac and found a 2 month old issue saying the issue was because the version needs bumped up to 24.2 (https://trac.macports.org/ticket/62326#no1). So I looked to see if anyone had already done this and saw this PR.

TLDR;

It would help if this was approved soon so I can install RabbitMQ. I figured I'd throw that out there that this PR could make a big difference if approved.

**Comment by herbygillot** [comment]
Normally a port is also listed as `openmaintainer`, which allows maintainers other than the original maintainer to apply changes and updates to the port in question.

In this instance, as per the this port is closed to only @ci42, so the standard procedure says we should wait for his review before merging in.

In this instance, since i have not seen @ci42 respond to much, I will go ahead and merge, but just to note, that's not as per the standard procedure.

**Comment by secsaba** [comment]
According to the guide https://guide.macports.org/#project.update-policies if the maintainer does not respond within 72 hours somebody else can commit the changes. We didn't reach that limit yet.

Thanks @herbygillot for the merge.

**Comment by cardiex-jgreene** [comment]
This code worked perfectly for me. Erlang installed and so did RabbitMQ! So I can confirm these two ports are working on Mac 12.1 Monterey with the Apple M1 Max chip. Thank you everyone.
